src/librbd/ImageCtx.h

```diff
@@ -177,6 +177,7 @@ namespace librbd {
     file_layout_t layout;
 
     cache::ImageCache<ImageCtx> *image_cache = nullptr;
+    bool discard_cache;
```

Comment:
Nit: this is a one time action not a permanent state for the image. Therefore, let's avoid adding this and instead the API should invoke an internal method to invalidate the cache.

Reply:
Removed from ImageCtx.